pkg/server/commands/reverseexpand/pipeline.go

```diff
@@ -571,162 +454,134 @@ func (r *baseResolver) resolve(senders []*sender, listeners []*listener) {
 	// Recursive senders will process infinitely until the parent worker's watchdog goroutine kills
 	// them.
 	wgRecursive.Wait()
+
+	span.SetAttributes(attribute.Int64("items.count", sentCount.Load()))
 }
 
-type edgeHandler func(context.Context, *Edge, []Item) iter.Seq[Item]
+type edgeHandler func(context.Context, *Edge, []string) iter.Seq[Item]
 
 type exclusionResolver struct {
-	reporter    Reporter
-	ctx         context.Context
+	reporter    track.Reporter
 	interpreter interpreter
-	status      *StatusPool
-	trk         tracker
 }
 
-func (r *exclusionResolver) resolve(senders []*sender, listeners []*listener) {
-	defer func() {
-		r.trk.Add(-1)
-		r.reporter.Report(false)
-	}()
+func (r *exclusionResolver) process(ctx context.Context, snd *sender, items *containers.Bag[Item], cleanup *containers.Bag[func()]) {
```

Comment:
The intersection and exclusion process methods are the same, should we consolidate them?